2016/proposals/result-aggregation-server-garg-prerit.md

```diff
@@ -0,0 +1,196 @@
+ 
+Contents 
+-------------- 
+1. **Abstract** 
+2. **Technical Details** 
+3. **Schedule of deliverables** 
+4. **Future Works** 
+5. **Open Source Development Experience** 
+6. **Academic Experience** 
+7. **Why this Project ?** 
+ 
+## Abstract 
+Software Carpentry has been teaching researchers in science, engineering, medicine, and related disciplines the computing skills they need to get more done in less time and with less pain. Several workshops are held in which students are required to setup their system by installing necessary packages. Currently software carpentry provides the installation-test scripts so that, students can check that they've successfully installed any software required by their workshop. But the results generated by the scripts are not collected anywhere. In this project a server is to be written for storing results of installation-testing scripts, and also enhancing the testing scripts by adding an option for users to collect results and send them to the server.  In this project, an API will be made so that, the results can be collected on the server, and thus can be figured out what all problems students usually face in installations. This will also help in making informed decisions about deprecating older versions of various dependencies or modernizing the lessons to catch up with current systems. 
+ 
+## Technical Details 
+I am breaking it down to several steps in which this project has to be approached : 
+### System Information collection 
+The installation-test script need to be modified to retrieve some useful information about the user and the environment in which he is running the installation-test-script. 
+* Operating System information, which includes : 
+  * System information ( Linux / windows / Mac etc ) 
+  * Distribution information ( redhat / debian / fedora etc) 
+  * Machine Architecture information ( x86 / x64 ) 
+  * Complete platform information 
+  * Uname ( System info like kernel-name, kernel-version etc ) 
+  * Version ( System’s release version ) 
+  * Machine information ( Machine type, e.g. 'i386' ) 
+  
+All of this information can be retrieved by using a library in python called "Platform", code of which I wrote in my [demo](https://github.com/prerit2010/DemoAPI) application. The code snippet : 
+``` 
+import platform 
+ 
+system_dist = str(platform.dist()) 
+system = platform.system() 
+machine = platform.machine() 
+system_platform = platform.platform() 
+uname = platform.uname() 
+version = platform.version() 
+``` 
+* Compilation environment used by user 
+ * Python Version 
+ * Python Build 
+ * Python Compiler 
+ * Python implementation ( e.g. ‘CPython’, ‘IronPython’, ‘Jython’, ‘PyPy’ ) 
+  
+All of which can be retrieved by the same library "platform" : 
+``` 
+import platform 
+ 
+python_version = platform.python_version() 
+python_build = platform.python_build() 
+python_compiler = platform.python_compiler() 
+python_implem = platform.python_implementation() 
+``` 
```

Comment:
I think exact Python version is enough – whatever installation-testing script gathers is probably fine.